src/OrthoViewController.cc

```diff
@@ -0,0 +1,289 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <ignition/common/Console.hh>
+
+#include "ignition/rendering/OrthoViewController.hh"
+#include "ignition/rendering/Scene.hh"
+#include "ignition/rendering/Visual.hh"
+
+/// \internal
+/// \brief OrthoViewController private data.
+class ignition::rendering::OrthoViewControllerPrivate
+{
+  /// \brief Build a custom scaled orthographic projection matrix.
+  /// \param[in] _left Left position
+  /// \param[in] _right Right position
+  /// \param[in] _bottom Bottom position
+  /// \param[in] _top Top position
+  /// \param[in] _near Near clip distance
+  /// \param[in] _far Far clip distance
+  public: math::Matrix4d BuildScaledOrthoMatrix(
+              float _left, float _right,
+              float _bottom, float _top,
+              float _near, float _far) const;
+
+  /// \brief Pointer to camera
+  public: CameraPtr camera;
+
+  /// \brief Scale used for zooming within the orthographic view
+  public: double scale = 1.0;
+
+  /// \brief Target point for camera movements
+  public: math::Vector3d target;
+};
+
+using namespace ignition;
+using namespace rendering;
+
+//////////////////////////////////////////////////
+OrthoViewController::OrthoViewController()
+    : dataPtr(new OrthoViewControllerPrivate)
+{
+  this->dataPtr->scale = 100;
+}
+
+//////////////////////////////////////////////////
+OrthoViewController::OrthoViewController(const CameraPtr &_camera)
+    : dataPtr(new OrthoViewControllerPrivate)
+{
+  this->dataPtr->scale = 100;
+  this->SetCamera(_camera);
+}
+
+//////////////////////////////////////////////////
+OrthoViewController::~OrthoViewController()
+{
+}
+
+//////////////////////////////////////////////////
+void OrthoViewController::SetCamera(const CameraPtr &_camera)
+{
+  this->dataPtr->camera = _camera;
+  if (!this->dataPtr->camera)
+    return;
+
+  // reset scale and update projection
+  if (this->dataPtr->camera->ProjectionType() != CPT_ORTHOGRAPHIC)
+  {
+    this->dataPtr->scale = 100;
+  }
+  this->dataPtr->camera->SetProjectionType(CPT_ORTHOGRAPHIC);
+  double width = this->dataPtr->camera->ImageWidth();
+  double height = this->dataPtr->camera->ImageHeight();
+  this->Resize(width, height);
+}
+
+//////////////////////////////////////////////////
+CameraPtr OrthoViewController::Camera() const
+{
+  return this->dataPtr->camera;
+}
+
+//////////////////////////////////////////////////
+void OrthoViewController::SetTarget(const math::Vector3d &_target)
+{
+  this->dataPtr->target = _target;
+}
+
+//////////////////////////////////////////////////
+math::Vector3d OrthoViewController::Target() const
```

Comment:
How about returning const ref so we don't force a copy?

Reply:
done. 6e2e25f